fuel-p2p/src/service.rs

```diff
@@ -141,19 +205,177 @@ impl FuelP2PService {
         request_id: RequestId,
         message: OutboundResponse,
     ) -> Result<(), ResponseError> {
-        self.swarm
-            .behaviour_mut()
-            .send_response_msg(request_id, message)
+        match (
+            self.network_codec.convert_to_intermediate(&message),
+            self.inbound_requests_table.remove(&request_id),
+        ) {
+            (Ok(message), Some(channel)) => {
```

Comment:
Is this purely a libp2p update? It looks like a lot of new logic was introduced here, the added logic and code looks good but just want to make sure that this is indeed added code

Reply:
Refer to the (updated) PR description for clarification.

Reply:
@xgreenx good point, created a separate issue for it https://github.com/FuelLabs/fuel-core/issues/669